nucleus/metrics/polygon_metrics.py

```diff
@@ -0,0 +1,213 @@
+        from nucleus.metrics import MetricResult, PolygonMetric
+        from nucleus.metrics.polygon_utils import BoxOrPolygonAnnotation, BoxOrPolygonPrediction
+
+        class MyPolygonMetric(PolygonMetric):
+            def eval(
+                self,
+                annotations: List[BoxOrPolygonAnnotation],
+                predictions: List[BoxOrPolygonPrediction],
+            ) -> MetricResult:
+                value = (len(annotations) - len(predictions)) ** 2
+                weight = len(annotations)
+                return MetricResult(value, weight)
+
+        box = BoxAnnotation(
+            label="car",
+            x=0,
+            y=0,
+            width=10,
+            height=10,
+            reference_id="image_1",
+            annotation_id="image_1_car_box_1",
+            metadata={"vehicle_color": "red"}
+        )
+
+        metric = MyPolygonMetric()
+        metric([box], [box])
+    """
+
+    def __init__(
+        self,
+        enforce_label_match: bool = False,
+        confidence_threshold: float = 0.0,
+    ):
+        """Initializes PolygonMetric abstract object.
+
+        Args:
+            enforce_label_match: whether to enforce that annotation and prediction labels must match. Default False
+            confidence_threshold: minimum confidence threshold for predictions. Must be in [0, 1]. Default 0.0
+        """
+        self.enforce_label_match = enforce_label_match
+        assert 0 <= confidence_threshold <= 1
+        self.confidence_threshold = confidence_threshold
+
+    @abstractmethod
+    def eval(
+        self,
+        annotations: List[BoxOrPolygonAnnotation],
+        predictions: List[BoxOrPolygonPrediction],
+    ) -> MetricResult:
+        # Main evaluation function that subclasses must override.
+        pass
+
+    def __call__(
+        self, annotations: AnnotationList, predictions: PredictionList
+    ) -> MetricResult:
+        if self.confidence_threshold > 0:
+            predictions = confidence_filter(
+                predictions, self.confidence_threshold
+            )
+        polygon_annotations: List[BoxOrPolygonAnnotation] = []
+        polygon_annotations.extend(annotations.box_annotations)
+        polygon_annotations.extend(annotations.polygon_annotations)
+        polygon_predictions: List[BoxOrPolygonAnnotation] = []
+        polygon_predictions.extend(predictions.box_predictions)
+        polygon_predictions.extend(predictions.polygon_predictions)
+
+        eval_fn = label_match_wrapper(self.eval)
+        result = eval_fn(
+            polygon_annotations,
+            polygon_predictions,
+            enforce_label_match=self.enforce_label_match,
+        )
+        return result
+
+
+class PolygonIOU(PolygonMetric):
+    """Calculates the average IOU between box or polygon annotations and predictions."""
+
+    def __init__(
+        self,
+        enforce_label_match: bool = False,
+        iou_threshold: float = 0.0,
```

Comment:
My thinking is that we can expose information about what defaults are used in Model CI in two ways:
1) Currently, we can keep these defaults in code so that users who want to see what the values are can find them directly here as defaults.
2) Once the UI shows the source code of evaluation functions, we can get rid of the defaults here and instead show them in the source code (e.g. `PolygonIOU(enforce_label_match=True, iou_threshold=0.2, confidence_threshold=0.5)`).

Reply:
In that case, I'll add a TODO here in `polygon_metrics.py` to remove the defaults once we can surface these defaults more directly to the user.